components/d2l-applied-filters/d2l-applied-filters.js

```diff
@@ -107,9 +110,13 @@ class D2lAppliedFilters extends RtlMixin(LocalizeStaticMixin(LitElement)) {
 
 	constructor() {
 		super();
+		this._clearFiltersClicked = this._clearFiltersClicked.bind(this);
 		this._clearSelected = this._clearSelected.bind(this);
+		this._handleDropdownChange = this._handleDropdownChange.bind(this);
 		this._setSelectedOptions = this._setSelectedOptions.bind(this);
 		this._update = this._update.bind(this);
+
+		this._hideClearFiltersButton = true;
```

Comment:
This line means that the “clear filters” button does not appear when the page first loads. I have assumed that the page will never load with 4+ filters already applied.

Reply:
If we handle the hiding at render time then we won't have to worry about this edge case

```
<d2l-button-subtle text="${this.localize('clearFilters')}" ?hidden="${this._selectedEntries.length < CLEAR_FILTERS_THRESHOLD}" @click="${this._clearFiltersClicked}"></d2l-button-subtle>
```
but you could also go one step further and do something like
```
shouldHideClearButton() {
}
...
${this.shouldHideClearButton(this._selectedEntries.length < CLEAR_FILTERS_THRESHOLD) ? html`<d2l-button-subtle text="${this.localize('clearFilters')}" @click="${this._clearFiltersClicked}"></d2l-button-subtle>` : ''}
```

Doing this logic at render is good because it can be evaluated purely from other properties

Reply:
I see what you're saying. However, `this._selectedEntries` is undefined at render time, so there would have to be another solution

Reply:
I think I have a good solution, I'll push in a minute

Reply:
I tested and pushed my change. Now "clear filters" will appear on page render if there are enough filters already applied.
I renamed my function `this._handleDropdownChange` to `this._refreshFilters` and I call it in the constructor